Improvement in RipUtils

Gaboso opened the pull request:
Defined a constant instead of duplicating this literal "imgur.com".
Defined a constant instead of duplicating this literal "http: //".
Moved strings to the left side of the string comparison.
Added a private constructor to hide the implicit public one.
Removed redundant initializers.
Changed indexOf(char) to single quotes for optimized for single characters.
Removed commented code blocks.
In the urlFromDirectoryName method, the first 'if' is always null, so I removed it.
Changed constant logger name to LOGGER.
Replace usage of System.out by Logger.

Gaboso committed:
Improvement in RipUtils

Defined a constant instead of duplicating this literal "imgur.com".
Defined a constant instead of duplicating this literal "http: //".
Moved strings to the left side of the string comparison.
Added a private constructor to hide the implicit public one.
Removed redundant initializers.
Changed indexOf(char) to single quotes for optimized for single characters.
Removed commented code blocks.
In the urlFromDirectoryName method, the first 'if' is always null, so I removed it.
Changed constant logger name to LOGGER.
Replace usage of System.out by Logger.

Gaboso committed:
Merge branch 'master' of https://github.com/4pr0n/ripme

# Conflicts:
#	src/main/java/com/rarchives/ripme/utils/RipUtils.java

rautamiekka commented:
While it's not 100% necessary, escaping the dots in website addresses in a RegEx ensures 100% success and is a performance optimization.

Also, just in case a site switches from HTTP to HTTPS or the other way around later a constant should be defined (instead of hardcoding in the code) and used like this (imitated from the very code) in Imgur's case:
```java
private static final String SITE_PROTOCOL = "http";
private static final String SITE_ADDRESS = SITE_PROTOCOL + "://imgur.com";

some_code(SITE_ADDRESS + "/01.png");
```

metaprime commented on src/main/java/com/rarchives/ripme/utils/RipUtils.java:
```diff
@@ -9,97 +22,91 @@
         List<URL> result = new ArrayList<URL>();
 
-        logger.debug("Checking " + url);
+        LOGGER.debug("Checking " + url);
```
Lowercase logger is used throughout ripme. Sure it's a convention for "constants" to be in all caps, but that static final variable isn't even benefitting from pretending to be a constant. Sure you can't reassign it, but as an output logger, it's state changes. I don't think anything useful is documented there. I'd prefer keeping it lowercase `logger` for consistency with the rest of the program and to reduce the size of unnecessary diffs. If you're serious about this change, it needs to change in the whole project.

metaprime commented on src/main/java/com/rarchives/ripme/utils/RipUtils.java:
```diff
@@ -1,5 +1,18 @@
+import org.apache.log4j.Logger;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
```
Just wondering for future consistency, what tool or scheme did you use to sort the imports?

Gaboso replied:
I'm using IntelliJ IDEA

metaprime commented on src/main/java/com/rarchives/ripme/utils/RipUtils.java:
```diff
@@ -9,97 +22,91 @@
         // Imgur album
-        if ((url.getHost().endsWith("imgur.com"))
-                && url.toExternalForm().contains("imgur.com/a/")) {
+        if ((url.getHost().endsWith(IMGUR_COM)) && url.toExternalForm().contains("imgur.com/a/")) {
```
Not going to replace the `imgur.com` at the end of this line?

metaprime commented on src/main/java/com/rarchives/ripme/utils/RipUtils.java:
```diff
@@ -109,38 +116,31 @@ else if (url.toExternalForm().contains("eroshare.com")) {
 
-        if (url.getHost().equals("imgur.com") ||
-                url.getHost().equals("m.imgur.com")) {
+        if (url.getHost().equals(IMGUR_COM) || "m.imgur.com".equals(url.getHost())) {
```
make a constant for M_IMGUR_COM

metaprime commented on src/main/java/com/rarchives/ripme/utils/RipUtils.java:
```diff
@@ -149,122 +149,156 @@ public static Pattern getURLRegex() {
+        String url = urlFromImgurDirectoryName(dir);
+
+        if (url == null)
+            url = urlFromImagefapDirectoryName(dir);
```
So this whole thing, as a style guideline one of three things can happen. `if` statement with a single statement body all on a single line (as it was before), the same with braces around the body (my preference), or indented WITH braces added around the body.

Absolutely please do not use the indented form without adding brackets.

IMO, the single-line forms are more readable in this case.

If you want to make a change here, please add brackets around the if bodies and it's up to you whether to reformat so the bodies are all on their own lines.

Also looking at this code now, I feel like there's got to be a better way of writing this bit of code (more efficient, etc.). If you make that sort of change, please do it is in a separate commit or pull request.

metaprime commented on src/main/java/com/rarchives/ripme/utils/RipUtils.java:
```diff
@@ -149,122 +149,156 @@ public static Pattern getURLRegex() {
-        if (!dir.startsWith(site + "_")) {
+        if (!dir.startsWith(site + "_"))
             return null;
-        }
```
Don't remove braces around if or loop bodies. They should be required AFAIC.

metaprime commented on src/main/java/com/rarchives/ripme/utils/RipUtils.java:
```diff
@@ -149,122 +149,156 @@ public static Pattern getURLRegex() {
-        if (!dir.startsWith("reddit_")) {
+        if (!dir.startsWith("reddit_"))
             return null;
-        }
```
Keep braces.

metaprime commented on src/main/java/com/rarchives/ripme/utils/RipUtils.java:
```diff
@@ -149,122 +149,156 @@ public static Pattern getURLRegex() {
+        if (NumberUtils.isDigits(directory))
+            url = "http://www.imagefap.com/gallery.php?gid=" + directory;
+        else
+            url = "http://www.imagefap.com/gallery.php?pgid=" + directory;
```
Please keep braces.

Gaboso committed:
Changed logger to maintain the application's current naming pattern.
Bringing back the braces.
Created constant for "m.imgur.com"
Removed line breaks in urlFromDirectory method.
Also created constant for initial url of imgur albums.

metaprime commented:
@Gaboso Why did you close the PR? Apologies for not following up on this.
